runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -566,14 +588,22 @@ public void clear() {
     @Override
     @Accessor
     public @Nonnull Set<Entry<K, V>> entrySet() {
-        List<Entry<K, V>> copy = new ArrayList<>(mainMap.size());
-        for (Map.Entry<K, V> entry : mainMap.entrySet()) {
-            copy.add(new AbstractMap.SimpleImmutableEntry<>(entry.getKey(),
-                    entry.getValue()));
-        }
-        return new ImmuableListSetWrapper<>(copy);
+        return mainMap.entryStream().map(entry ->
+                new AbstractMap.SimpleImmutableEntry<>(entry.getKey(), entry.getValue()))
+                .collect(ImmutableSet.toImmutableSet());
```

Comment:
This the objects have to be rehashed this can be expensive ~5-10x 

For a map of 1 million entries it:
1 million

iter 0
[guava] elapsed(ms) :                  463
[ImmuableListSetWrapper] elapsed(ms) : 61
iter 1
[guava] elapsed(ms) :                  402
[ImmuableListSetWrapper] elapsed(ms) : 63
iter 2
[guava] elapsed(ms) :                  388
[ImmuableListSetWrapper] elapsed(ms) : 52
iter 3
[guava] elapsed(ms) :                  394
[ImmuableListSetWrapper] elapsed(ms) : 54


For a map of 500 thousand entries:
iter 0
[guava] elapsed(ms) :                  233
[ImmuableListSetWrapper] elapsed(ms) : 33
iter 1
[guava] elapsed(ms) :                  184
[ImmuableListSetWrapper] elapsed(ms) : 38
iter 2
[guava] elapsed(ms) :                  197
[ImmuableListSetWrapper] elapsed(ms) : 31


In this experiment, the map uses a UUID for both key and value, but for complex types, this can be much more expensive.

Reply:
Can you share you benchmarking code? Previously we were doing `copy.add(...)`, so I am not sure I am able to explain the difference.

Reply:
I see. `ImmuableListSetWrapper` is our implementation.

Reply:
Alright, I will collect it to a `List` and convert it to `ImmuableListSetWrapper`, so there should not be any performance difference.

Reply:
Fixed.